Some opaque values are not accepted by leshan when using LWM2M with senml json

### Describe the bug

When using LWM2M with senml/json, opaque values are send base64 encoded. Not all send values are accepted by the leshan-server. The leshan-server seems to expect base64 urlsafe-encoded with padding omitted. 
This seems to be the intended way, as described here:
https://github.com/OpenMobileAlliance/OMA_LwM2M_for_Developers/issues/553
Currently zephyr sends the opaque values in senml/json as base64 encoded (not urlsafe) with padding.

### Regression

- [ ] This is a regression.

### Steps to reproduce

- Add an resource with opaque value to the lwm2m client sample
	- For this test, I added a binary data-object to the sample. (shown here: https://github.com/jbr-ia/zephyr/commit/907c5fb92afcaa3af2531ddf3e4872cfd521ca49)
- Compile and run ```west build -b qemu_cortex_a53 -- -DOVERLAY_CONFIG="overlay-data-cache.conf;overlay-dtls.conf;overlay-lwm2m-1.1.conf;overlay-lwm2m-1.1-core-interop.conf;overlay-queue.conf;overlay-tickless.conf"```
- run the sample ```west build -t run```
- On the leshan-server, set single and multi-value to senml_json 
- On the leshan server, Enable observe for the binary data object data, resource 19/0/0
- in the zephyr-terminal, write a value to the resource
  - ```lwm2m write /19/0/0 -s "123"``` and see the value changes in leshan
  - ```lwm2m write /19/0/0 -s "1"``` and see the value does not change in leshan (because there is padding)
  - ```lwm2m write /19/0/0 -s "???"``` and see the value does not change in leshan (because there is a '/' in the base64 encoded string which should not be in the base64 urlsafe variant)
  - ```lwm2m write /19/0/0 -s "~~~"```and see the value does not change in leshan (because there is a '+' in the base64 encoded string which should not be in the base64 urlsafe variant)
  - ```lwm2m write /19/0/0 -s "123456"``` and see communication still works (because with this string there is no difference between base64 and base64 url-safe encoding)


If I change the encoded value to urlsafe without padding before sending (as done here: https://github.com/jbr-ia/zephyr/commit/28e71cc2260ce7a1c1cdea62dc0ec517e5e06280) , than all values seem to be received correctly on leshan.


### Relevant log output

```shell
the leshan-log shows output like:

Caused by: org.eclipse.leshan.core.util.json.JsonException: Node vd with value 'Pz8/' is not in valid Base64 format.
	at .....
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

OS: Ubuntu 24.04 (in WSL)
Version used: main (commit 7e3c5ab2bb2a9f4bf1b0afd43c872e6f19461777) with some changes as described in 'steps to reproduce'

### Additional Context

_No response_

--- a/subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c
+++ b/subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c
@@ -905,8 +905,28 @@ static int json_append_bytes_base64(const char *bytes, size_t len, void *data)
 			/* No space available for base64 data */
 			return -ENOMEM;
 		}
+
+		/* Change base64 data to URL-safe BASE64 data without padding */
+		int padding_removed = 0;
+
+		for (int i = 0; i < temp_length; i++) {
+			switch (CPKT_BUF_W_PTR(out->out_cpkt)[i]) {
+			case '+':
+				CPKT_BUF_W_PTR(out->out_cpkt)[i] = '-';
+				break;
+			case '/':
+				CPKT_BUF_W_PTR(out->out_cpkt)[i] = '_';
+				break;
+			case '=':
+				CPKT_BUF_W_PTR(out->out_cpkt)[i] = 0;
+				padding_removed++;
+				break;
+			default:
+				break;
+			}
+		}
 		/* Update Data offset */
-		out->out_cpkt->offset += temp_length;
+		out->out_cpkt->offset += (temp_length - padding_removed);
 	} else {
 		if (buf_append(CPKT_BUF_WRITE(fd->out->out_cpkt), bytes, len) < 0) {
 			return -ENOMEM;

